src/env/tock/mod.rs

```diff
@@ -18,10 +21,46 @@ use rng256::TockRng256;
 
 mod storage;
 
+pub struct TockCtapHidChannel {
+    transport: Transport,
+}
+
+impl CtapHidChannel for TockCtapHidChannel {
+    fn send_or_recv_with_timeout(
+        &mut self,
+        buf: &mut [u8; 64],
+        timeout: CtapDuration,
+    ) -> SendOrRecvResult {
+        let endpoint = match self.transport {
+            Transport::MainHid => usb_ctap_hid::UsbEndpoint::MainHid,
+            #[cfg(feature = "vendor_hid")]
+            Transport::VendorHid => usb_ctap_hid::UsbEndpoint::VendorHid,
+        };
+        match usb_ctap_hid::send_or_recv_with_timeout(
+            buf,
+            timer::Duration::from_ms(timeout.integer() as isize),
+            endpoint,
+        ) {
+            Ok(usb_ctap_hid::SendOrRecvStatus::Timeout) => Ok(SendOrRecvStatus::Timeout),
+            Ok(usb_ctap_hid::SendOrRecvStatus::Sent) => Ok(SendOrRecvStatus::Sent),
+            Ok(usb_ctap_hid::SendOrRecvStatus::Received(recv_endpoint))
+                if endpoint == recv_endpoint =>
+            {
+                Ok(SendOrRecvStatus::Received)
+            }
+            _ => Err(SendOrRecvError),
```

Comment:
Interesting, if we're receiving something on the other channel, we're dropping it. This might be problematic. Not sure what's the best solution with the current design. @kaczmarczyck any ideas?